### PR TITLE
refactor(components): [container] switch to script-setup syntax

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/container/src/container.vue [hunk 1]
```diff
@@ -3,43 +3,39 @@
...
-  props: {
-    direction: {
-      type: String,
-      default: '',
-    },
+})
+const props = defineProps({
+  direction: {
+    type: String,
+    default: '',
```
Comment (JeremyWuuuuu) [inline packages/components/container/src/container.vue:18]: **Nitpick:** remove unnecessary `default: ''` since this is a `String` type.

**Review by jw-foss — COMMENTED** [review]
Left a comment 🏓

File: packages/components/container/src/container.vue (hunk 1 above)
Comment (btea) [reply packages/components/container/src/container.vue]: Done  🏓

**Comment by JeremyWuuuuu** [comment]
LGTM. Approved

**Review by jw-foss — APPROVED** [review]

**Pull request merged**